### PR TITLE
Revert "Merge pull request #2657 from ipfs/feature/add-defaults-to-add"

### PR DESCRIPTION
**Pull request by RichardLitt**
I think I did this incorrectly, but I am not sure. Why should a revert have merge conflicts? This is an attempt at fixing #3164.

### CODE REVIEW

File: core/commands/add.go [hunk 1]
```diff
@@ -69,26 +67,29 @@ You can now refer to the added file in a gateway, like so:
...
-		cmds.BoolOption(onlyHashOptionName, "n", "Only chunk and hash - do not write to disk.").Default(false),
-		cmds.BoolOption(wrapOptionName, "w", "Wrap files with a directory object.").Default(false),
-		cmds.BoolOption(hiddenOptionName, "H", "Include files that are hidden. Only takes effect on recursive add.").Default(false),
+		cmds.BoolOption(trickleOptionName, "t", "Use trickle-dag format for dag generation."),
+		cmds.BoolOption(onlyHashOptionName, "n", "Only chunk and hash - do not write to disk."),
+		cmds.BoolOption(wrapOptionName, "w", "Wrap files with a directory object."),
+		cmds.BoolOption(hiddenOptionName, "H", "Include files that are hidden. Only takes effect on recursive add."),
 		cmds.StringOption(chunkerOptionName, "s", "Chunking algorithm to use."),
-		cmds.BoolOption(pinOptionName, "Pin this object when adding.").Default(true),
+		cmds.BoolOption(pinOptionName, "Pin this object when adding.  Default: true."),
```
Comment (kevina) [inline core/commands/add.go]: This should still be ".Default(true)."

File: core/commands/add.go [hunk 2]
```diff
@@ -134,7 +135,11 @@ You can now refer to the added file in a gateway, like so:
 		hidden, _, _ := req.Option(hiddenOptionName).Bool()
 		silent, _, _ := req.Option(silentOptionName).Bool()
 		chunker, _, _ := req.Option(chunkerOptionName).String()
-		dopin, _, _ := req.Option(pinOptionName).Bool()
+		dopin, pin_found, _ := req.Option(pinOptionName).Bool()
+
+		if !pin_found { // default
+			dopin = true
+		}
```
Comment (kevina) [inline core/commands/add.go]: This `if !pin_found {...}` should not be necessary.  See earlier comment.

File: core/commands/add.go [hunk 3]
```diff
@@ -69,26 +67,29 @@ You can now refer to the added file in a gateway, like so:
...
+		// ipfs cli progress bar defaults to true
+		progress, found, _ := req.Option(progressOptionName).Bool()
 		if !found {
-			req.SetOption(progressOptionName, true)
+			progress = true
 		}
 
+		req.SetOption(progressOptionName, progress)
+
 		sizeFile, ok := req.Files().(files.SizeFile)
```
Comment (kevina) [inline core/commands/add.go]: Note: I don't understand what is going on here.  Why are we setting progressOptionName if it is not found.  See my note in the PostRun function.

File: core/commands/add.go [hunk 4]
```diff
@@ -242,7 +247,7 @@ You can now refer to the added file in a gateway, like so:
 			return
 		}
 
-		progress, _, err := req.Option(progressOptionName).Bool()
+		progress, prgFound, err := req.Option(progressOptionName).Bool()
 		if err != nil {
```
Comment (kevina) [inline core/commands/add.go]: Note: Okay so how is it that prgFound can be false, it we explicitly set it the PreRun.  I am very unsure of the correct behavior here.  What was the original intention.  @whyrusleeping @Kubuxu any ideas?

**Comment by kevina** [comment]
Everyone.  I will be willing to just take over this pull request and redo it if it would be easier, but I don't want to step on anyone's toes.

**Comment by RichardLitt** [comment]
My toes are very small. It is hard to step on them. :) 

Take it away! Thank you, @kevina.

**Comment by RichardLitt** [comment]
Regarding your questions: I don't know, I am not invested, and I won't pretend my edits were high quality. They were the best I managed at the time, but realistically, if you don't understand something I did, that's my fault, not yours. Feel free to close this PR unmerged if that is the easiest way forward.

**Comment by kevina** [comment]
@RichardLitt, Okay, I will just redo this in the next day or so.

**Comment by kevina** [comment]
@RichardLitt okay I redid the revert in #3464.

Note that your revert was fine, when I did it I got the same results.  I just chose to resolve a conflict slightly differently.

**Comment by RichardLitt** [comment]
Thank you, @kevina!

**Pull request closed without merging**